cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMirrorMaker2AssemblyOperator.java

```diff
@@ -358,6 +362,47 @@ private static void addClusterToMirrorMaker2ConnectorConfig(Map<String, Object>
         config.putAll(cluster.getAdditionalProperties());
     }
 
+    private Future<Map<String, Object>> reconcileMirrorMaker2Connector(Reconciliation reconciliation, KafkaMirrorMaker2 mirrorMaker2, KafkaConnectApi apiClient, String host, String connectorName, KafkaConnectorSpec connectorSpec, KafkaMirrorMaker2Status mirrorMaker2Status) {
+        return apiClient.createOrUpdatePutRequest(host, KafkaConnectCluster.REST_API_PORT, connectorName, asJson(connectorSpec))
+                .compose(ignored -> apiClient.statusWithBackOff(new BackOff(200L, 2, 6), host, KafkaConnectCluster.REST_API_PORT, connectorName))
+                .compose(status -> {
+                    Object path = ((Map) status.getOrDefault("connector", emptyMap())).get("state");
+                    if (!(path instanceof String)) {
+                        return Future.failedFuture("JSON response lacked $.connector.state");
+                    } else {
+                        String state = (String) path;
+                        boolean shouldPause = Boolean.TRUE.equals(connectorSpec.getPause());
+                        if ("RUNNING".equals(state) && shouldPause) {
+                            log.debug("{}: Pausing connector {}", reconciliation, connectorName);
+                            return apiClient.pause(host, KafkaConnectCluster.REST_API_PORT, connectorName)
+                                    .compose(ignored ->
+                                            apiClient.status(host, KafkaConnectCluster.REST_API_PORT,
+                                                    connectorName));
+                        } else if ("PAUSED".equals(state) && !shouldPause) {
+                            log.debug("{}: Resuming connector {}", reconciliation, connectorName);
+                            return apiClient.resume(host, KafkaConnectCluster.REST_API_PORT,
+                                    connectorName)
+                                    .compose(ignored ->
+                                            apiClient.status(host, KafkaConnectCluster.REST_API_PORT,
+                                                    connectorName));
+
```

Comment:
This looks to be more or less identical to the code we use for `KafkaConnectors`. Can we factor out a common method?

Reply:
Agreed - the latest commit factors out the common code.